exercises/strain/strain.example.ts

```diff
@@ -0,0 +1,23 @@
+interface Predicate<T> {
+  (element: T): boolean
+}
+
+export function keep<T>(array: T[], predicate: Predicate<T>): T[] {
+  const output: T[] = []
+  array.forEach((el: T) => {
+    if (predicate(el)) {
+      output.push(el)
+    }
+  })
+  return output
+}
+
+export function discard<T>(array: T[], predicate: Predicate<T>): T[] {
+  const output: T[] = []
+  array.forEach((el: T) => {
```

Comment:
What do you think? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter

Reply:
My original implementation was done with two lines.

For keep:
```
array.filter(predicate)
```

For discard:
```
array.filter((el) => !predicate(el)
```

Although, I realized that the `ecmascript` exercise prompt said specifically not to use filter. I don't feel strongly either way, just wanted to follow the rules of the exercise.

Reply:
Ah yeah. okay!